lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -0,0 +1,169 @@
+# returned by boto3's GCE describe_instances method.
+class GcpMockModule(object):
+    def __init__(self, params):
+        self.params = params
+
+    def fail_json(self, *args, **kwargs):
+        raise AnsibleError(kwargs['msg'])
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'gcp_compute'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        self.group_prefix = 'gcp_'
+
+    def _populate(self, resp):
+        '''
+            :param resp: A JSON response
+        '''
+        for item in resp:
+            hostname = item['name']
+            self.inventory.add_host(hostname)
+            for key in item:
+                self.inventory.set_variable(hostname, key, item[key])
+            self.inventory.add_child('all', hostname)
+
+    def _validate_config(self, loader, path):
+        '''
+            :param loader: an ansible.parsing.dataloader.DataLoader object
+            :param path: the path to the inventory config file
+            :return the contents of the config file
+        '''
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith('.gcp.yml') or path.endswith('.gcp.yaml'):
+                return self._read_config_data(path)
+        raise AnsibleParserError("Not a gce inventory plugin configuration file")
+
+    def self_link(self, params):
+        '''
+            :param params: a dict containing all of the fields relevant to build URL
+            :return the formatted URL as a string.
+        '''
+        return "https://www.googleapis.com/compute/v1/projects/{project}/zones/{zone}/instances".format(**params)
+
+    def fetch_list(self, params, link, query):
+        '''
+            :param params: a dict containing all of the fields relevant to build URL
+            :param link: a formatted URL
+            :param query: a formatted query string
+            :return the JSON response containing a list of instances.
+        '''
+        module = GcpMockModule(params)
+        auth = GcpSession(module, 'compute')
+        response = auth.get(link, params={'filter': query})
+        return self.return_if_object(module, response)
+
+    def _get_query_options(self, config_data):
+        '''
+            :param config_data: contents of the inventory config file
+            :return A string
+        '''
+        queries = []
+        for obj_key in config_data['filters']:
+            for conditional in config_data['filters'][obj_key]:
+                if conditional not in ['ne', 'eq']:
+                    raise AnsibleParserError("All conditionals must be eq or ne")
+                for item in config_data['filters'][obj_key][conditional]:
+                    query = [obj_key, conditional, item]
+                    queries.append("(%s)" % ' '.join(query))
+
+        return ' '.join(queries)
+
+    def return_if_object(self, module, response):
```

Comment:
I think this should be `_return_if_object` since it isn't part of the public API.